tests/lowering/reduction/test_sum.py

```diff
@@ -0,0 +1,62 @@
+import torch
+import torch_ttnn
+import pytest
+import ttnn
+
+from tests.utils import assert_with_pcc
+
+
+class SumDimModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, input, dim, keepdim=False):
+        return torch.sum(input, dim, keepdim=keepdim)
+
+
+@pytest.mark.parametrize(
+    "input_shape, dim",
+    (
+        ((1, 32, 32), (-1, -2)),
+        ((1, 1, 768), (0, 1)),
+        ((1, 1000), (0,)),
+        ((1, 1024, 256), (0, 1)),
+        ((1, 1024, 7, 7), (2, 3)),
+        ((1, 12, 16), (1,)),
+        ((1, 12, 16), (2,)),
+        ((1, 512), (1,)),
+        ((1, 64), (0,)),
+        ((1024, 160), (0,)),
+        ((1024, 640), (0,)),
+        ((14, 2048), (0,)),
+        ((14, 512), (0,)),
+        ((16384, 128), (0,)),
```

Comment:
Found mild PCC error locally:
```
expected_pytorch_result = tensor([[  53.5000,  -96.0000, -113.5000, -139.0000,  -72.5000,   39.2500,
         -159.0000,  -13.6875,  -87.0000,  ...187.0000, -130.0000,  -23.7500,  -18.7500,  -25.2500, -132.0000,
          -22.0000,   87.5000]], dtype=torch.bfloat16)
actual_pytorch_result = TorchTensor([[  63.0000, -111.5000, -129.0000, -142.0000,  -74.0000,   59.2500,
              -153.0000,  -14.1875,  -...000, -137.0000,   -5.2188,  -18.7500,  -34.0000, -153.0000,
               -15.6250,  102.5000]], dtype=torch.bfloat16)
pcc = 0.999

    def assert_with_pcc(expected_pytorch_result, actual_pytorch_result, pcc=0.999):
        assert list(expected_pytorch_result.shape) == list(
            actual_pytorch_result.shape
        ), f"list(expected_pytorch_result.shape)={list(expected_pytorch_result.shape)} vs list(actual_pytorch_result.shape)={list(actual_pytorch_result.shape)}"
        pcc_passed, pcc_message = comp_pcc(expected_pytorch_result, actual_pytorch_result, pcc)
>       assert pcc_passed, construct_pcc_assert_message(pcc_message, expected_pytorch_result, actual_pytorch_result)
E       AssertionError: 0.9923741703805607

tests/utils.py:219: AssertionError
```

Reply:
PCC accuracy declines as tensors grow in size because of aggregation of differences. Would it make sense to lower the threshold for tensors where a dimension is over 10k?